### PR TITLE
Add Security and Privacy Considerations sections

### PR DESCRIPTION
**Pull request by dkg**
We didn't have any explicit sections describing the concrete concerns
that we hope will be mitigated by these constructs, or any text about
what we are deliberately not covering here.

Hopefully this begins to address these concerns.

### CODE REVIEW

File: draft-protected-headers.md [hunk 1]
```diff
@@ -497,6 +497,37 @@ IANA Considerations
...
+This document describes a technique that can be used to defend against two security vulnerabilities in traditional end-to-end encrypted e-mail.
+
+Subject Leak
+------------
+
+While e-mail structure considers the Subject header to be part of the message metadata, nearly all users consider the Subject header to be part of the message content.
+
+As such, a user sending end-to-end encrypted e-mail may inadvertently leak sensitive material in the Subject line.
+
+If the user's MUA uses Protected Headers and obscures the Subject header as described in {{encrypted-subject}} then this breach of confidentiality
```
Comment (BjarniRunar) [inline draft-protected-headers.md]: Truncated paragraph: "then this breach of confidentiality" .... is avoided?

**Review by BjarniRunar — APPROVED** [review]
Truncated paragraph aside, this looks great to me.

**Comment by dkg** [comment]
Thanks for catching that screwed up sentence.  I've just cleaned up the branch to push a fix for it.

**Review by juga0 — APPROVED** [review]
lgtm

**Pull request merged**